hdk/jack_analyzer/syntax.py

```diff
@@ -0,0 +1,355 @@
+        _add_child(element, "keyword", "if")
+        _add_child(element, "symbol", "(")
+        element.appendChild(self.expression.to_xml(dom_tree))
+        _add_child(element, "symbol", ")")
+        _add_child(element, "symbol", "{")
+        element.appendChild(self.if_statements.to_xml(dom_tree))
+        _add_child(element, "symbol", "}")
+        if self.else_statements is not None:
+            _add_child(element, "keyword", "else")
+            _add_child(element, "symbol", "{")
+            element.appendChild(self.else_statements.to_xml(dom_tree))
+            _add_child(element, "symbol", "}")
+        return element
+
+
+@dataclass(frozen=True)
+class WhileStatement:
+    expression: Expression
+    statements: Statements
+
+    def to_xml(self, dom_tree: Document) -> Element:
+        element = dom_tree.createElement("whileStatement")
+        _add_child(element, "keyword", "while")
+        _add_child(element, "symbol", "(")
+        element.appendChild(self.expression.to_xml(dom_tree))
+        _add_child(element, "symbol", ")")
+        _add_child(element, "symbol", "{")
+        element.appendChild(self.statements.to_xml(dom_tree))
+        _add_child(element, "symbol", "}")
+        return element
+
+
+Statement: TypeAlias = ReturnStatement | DoStatement | LetStatement | IfStatement | WhileStatement
+
+
+@dataclass(frozen=True)
+class Statements:
+    statements: list[Statement]
+
+    def to_xml(self, dom_tree: Document) -> Element:
+        element = dom_tree.createElement("statements")
+        for statement in self.statements:
+            element.appendChild(statement.to_xml(dom_tree))
+        return element
+
+
+@dataclass(frozen=True)
+class VarDeclaration:
+    type_: str
+    names: list[str]
+
+    def to_xml(self, dom_tree: Document) -> Element:
+        element = dom_tree.createElement("varDec")
+        _add_child(element, "keyword", "var")
+        element_type = (
+            "keyword"
+            if self.type_ in {"int", "char", "boolean"}
+            else "identifier"
+        )
+        _add_child(element, element_type, self.type_)
+        for i, var_name in enumerate(self.names):
+            _add_child(element, "identifier", var_name)
+            if i < len(self.names) - 1:
+                _add_child(element, "symbol", ",")
+        _add_child(element, "symbol", ";")
+        return element
+
+
+@dataclass(frozen=True)
+class SubroutineBody:
+    var_declarations: list[VarDeclaration]
+    statements: Statements
+
+    def to_xml(self, dom_tree: Document) -> Element:
+        element = dom_tree.createElement("subroutineBody")
+        _add_child(element, "symbol", "{")
+        for var_dec in self.var_declarations:
+            element.appendChild(var_dec.to_xml(dom_tree))
+        element.appendChild(self.statements.to_xml(dom_tree))
+        _add_child(element, "symbol", "}")
+        return element
+
+
+@dataclass(frozen=True)
+class SubroutineDeclaration:
+    type_: [str, str]
```

Comment:
Split type into two attributes: type_: str, return_type: str, so you will not need to write self.type_[1]